src/runtime-rs/crates/hypervisor/src/device/device_manager.rs

```diff
@@ -0,0 +1,338 @@
+// Copyright (c) 2019-2022 Alibaba Cloud
+// Copyright (c) 2019-2022 Ant Group
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use crate::DeviceType::{Block, Undefined, Vfio};
+use crate::{
+    device::blk_dev_manager::BlockDeviceManager,
+    device_type::{Device, DeviceArgument, GenericConfig},
+    DeviceManagerInner, DeviceType, Hypervisor,
+};
+use anyhow::{anyhow, Context, Result};
+use ini::Ini;
+use std::{collections::HashMap, sync::Arc};
+use tokio::sync::{Mutex, RwLock};
+
+pub type ArcBoxDevice = Arc<Mutex<Box<dyn Device>>>;
+const SYS_DEV_PREFIX: &str = "/sys/dev";
+
```

Comment:
I think the definition below belongs to the block device manager.

Reply:
fixed